src/main/java/com/mongodb/hibernate/jdbc/MongoStatement.java

```diff
@@ -99,9 +99,21 @@ static List<String> getFieldNamesFromProjectStage(BsonDocument projectStage) {
         return fieldNames;
     }
 
-    private static boolean trueOrOne(BsonValue value) {
-        return (value.isBoolean() && value.asBoolean().getValue())
-                || (value.isNumber() && value.asNumber().intValue() == 1);
+    private static boolean isExcludeProjectSpecification(Map.Entry<String, BsonValue> specification) {
+        var key = specification.getKey();
+        var value = specification.getValue();
+        var exclude = (value.isBoolean() && !value.asBoolean().getValue())
+                || (value.isNumber() && value.asNumber().intValue() == 0);
+        if (exclude && !key.equals(ID_FIELD_NAME)) {
+            throw new RuntimeException(format(
+                    "Exclusions are not allowed in `$project` specifications, except for the [%s] field: [%s, %s]",
+                    ID_FIELD_NAME, key, value));
+        }
+        if (!value.isBoolean() && !value.isInt32()) {
```

Comment:
should we use `!value.isNumber()` to align with the above logic?

Reply:
Absolutely. Sorry for this nonsense. Done in https://github.com/mongodb/mongo-hibernate/pull/72/commits/075bef0bdd5b5a0aa3f23f6d3b4e284a2d5e738e.